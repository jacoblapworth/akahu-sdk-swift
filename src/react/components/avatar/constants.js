/**
 * @public
 * Map of sizes to XUI Classes properties
 */
export const sizeClassNames = {
	xsmall: 'xui-avatar-xsmall',
	small: 'xui-avatar-small',
	medium: '',
	large: 'xui-avatar-large',
	xlarge: 'xui-avatar-xlarge',
	xxlarge: 'xui-avatar-xxlarge'
};

/**
 * @public
 * Map of core class names
 */
export const classNames = {
	base: 'xui-avatar',
	counter: 'xui-avatar-counter',
	group: 'xui-avatargroup'
};

/**
 * @public
 * Map of color classes
 */
export const colorClassNames = [];
for(let i = 1; i <= 10; i++) {
	colorClassNames.push(`xui-avatar-color-${i}`);
}

/**
 * @public
 * Map of variants to class names
 */
export const variantClassNames = {
	business: 'xui-avatar-business'
};
