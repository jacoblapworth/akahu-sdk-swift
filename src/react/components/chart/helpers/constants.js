import {ns} from "../../helpers/xuiClassNamespace";

export const NAME_SPACE = `${ns}-chart`;
export const CHART_HEIGHT = 400;
export const CHART_WIDTH = 200;
export const X_AXIS_HEIGHT = 50;
export const Y_AXIS_WIDTH = 50;
export const BAR_MIN_WIDTH = 34;
export const BAR_MAX_WIDTH = 200;
