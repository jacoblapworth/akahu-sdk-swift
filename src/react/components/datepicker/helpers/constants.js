export const englishMonths = [
	'January',
	'February',
	'March',
	'April',
	'May',
	'June',
	'July',
	'August',
	'September',
	'October',
	'November',
	'December',
];

export const customClassNames = {
  container: 'xui-datepicker',
  interactionDisabled: 'xui-is-disabled',
  navBar: 'xui-datepicker--header',
  navButtonPrev: 'xui-datepicker--arrow-icon',
  navButtonNext: 'xui-datepicker--arrow-icon',
  month: 'xui-datepicker--body',
  caption: '',
  weekdays: '',
  weekdaysRow: 'xui-datepicker--weekdays',
  weekday: 'xui-datepicker--weekday',
  body: '',
  week: 'xui-datepicker--week',
  day: 'xui-datepicker--day',
  today: 'xui-datepicker--day-is-today',
  selected: 'xui-datepicker--day-is-selected',
  outside: 'xui-datepicker--day-is-othermonth',
  disabled: 'xui-datepicker--day-is-disabled',
};
