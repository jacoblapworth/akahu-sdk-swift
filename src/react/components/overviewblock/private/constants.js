const overviewSentiments = {
  standard: '',
  positive: 'positive',
  negative: 'negative',
  muted: 'muted',
};

module.exports = {
  overviewSentiments,
};
