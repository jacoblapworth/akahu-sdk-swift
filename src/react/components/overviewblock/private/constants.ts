// eslint-disable-next-line import/prefer-default-export
export const overviewSentiments = {
  standard: '',
  positive: 'positive',
  negative: 'negative',
  muted: 'muted',
};
