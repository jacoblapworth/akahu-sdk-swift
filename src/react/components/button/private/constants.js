import { ns } from '../../helpers/xuiClassNamespace';

export const baseClassName = `${ns}-button`;

/**
 * @public
 * Map of sizes to XUI Classes properties
 */
export const sizeClassNames = {
  xsmall: `${baseClassName}-xsmall`,
  small: `${baseClassName}-small`,
  medium: `${baseClassName}-medium`,
};

/**
 * @public
 * Map of widths to XUI Classes properties
 */
export const widthClassNames = {
  always: `${baseClassName}-fullwidth`,
  'small-down': `${baseClassName}-fullwidth-layout`,
  never: '',
};

/**
 * @public
 * Map of sizes to XUI Classes properties for icons
 */
export const iconSizeClassNames = {
  xsmall: `${baseClassName}-icon-xsmall`,
  small: `${baseClassName}-icon-small`,
  medium: `${baseClassName}-icon-medium`,
};

/**
 * @public
 * Map of ‘solid’, non-icon button class names
 */
export const standardVariantClassNames = {
  standard: `${baseClassName}-standard`,
  primary: `${baseClassName}-main`,
  create: `${baseClassName}-create`,
  negative: `${baseClassName}-negative`,
};

/**
 * @public
 * Map of icon button class names
 */
export const iconVariantClassNames = {
  icon: `${baseClassName}-icon`,
  'icon-inverted': `${baseClassName}-icon ${baseClassName}-icon-inverted`,
};

/**
 * @public
 * Map of borderless button class names
 */
export const borderlessVariantClassNames = {
  'borderless-standard': `${baseClassName}-borderless-standard`,
  'borderless-primary': `${baseClassName}-borderless-main`,
  'borderless-create': `${baseClassName}-borderless-create`,
  'borderless-negative': `${baseClassName}-borderless-negative`,
  'borderless-inverted': `${baseClassName}-borderless-inverted`,
  'borderless-muted': `${baseClassName}-borderless-muted`,
};

/**
 * @public
 * Map of button variants excluding icon variants
 */
export const textButtonVariants = {
  ...standardVariantClassNames,
  ...borderlessVariantClassNames,
  unstyled: '',
};

/**
 * @public
 * Map of all button variants
 */
export const buttonVariants = {
  ...textButtonVariants,
  ...iconVariantClassNames,
};

/**
 * @public
 * Map of button types to class names
 */
export const buttonTypes = {
  submit: 'submit',
  button: 'button',
  reset: 'reset',
};
