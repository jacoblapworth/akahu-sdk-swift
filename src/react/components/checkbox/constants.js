import { ns } from '../helpers/xuiClassNamespace';
/* eslint-disable import/prefer-default-export */
export const baseClass = `${ns}-styledcheckboxradio`;
