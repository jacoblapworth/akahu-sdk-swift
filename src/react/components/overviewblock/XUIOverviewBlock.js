import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { ns } from '../helpers/xuiClassNamespace';

const baseClass = `${ns}-overview`;

const XUIOverviewBlock = ({
  children,
  className,
  hasBackground,
  hasBorder,
  hasLayout,
  qaHook,
  textAlignment,
  ...spreadProps
}) => {
  const classes = cn(
    baseClass,
    className,
    hasBorder && `${baseClass}-has-border`,
    hasBackground && `${baseClass}-solid`,
    `${baseClass}-text-align-${textAlignment}`,
    hasLayout && `${baseClass}-layout`,
  );

  return (
    <div {...spreadProps} className={classes} data-automationid={qaHook}>
      {children}
    </div>
  );
};

export default XUIOverviewBlock;

XUIOverviewBlock.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  /** Whether the block should have a solid background. Defaults to true */
  hasBackground: PropTypes.bool,
  /** Whether to show wrapping border on the entire block. Defaults to true */
  hasBorder: PropTypes.bool,
  /**
   * Applies default layout styling.
   */
  hasLayout: PropTypes.bool,
  qaHook: PropTypes.string,
  /** How to align text, generally, across all sections */
  textAlignment: PropTypes.oneOf(['left', 'center', 'right']),
};

XUIOverviewBlock.defaultProps = {
  hasBackground: true,
  hasBorder: true,
  hasLayout: true,
  textAlignment: 'center',
};
