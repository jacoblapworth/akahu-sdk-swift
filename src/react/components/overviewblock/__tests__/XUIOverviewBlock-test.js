import React from 'react';
import { axe, toHaveNoViolations } from 'jest-axe';
import XUIOverviewBlock from '../XUIOverviewBlock';
import XUIOverviewSection from '../XUIOverviewSection';
import { overviewSentiments } from '../private/constants';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });
expect.extend(toHaveNoViolations);

describe('<XUI OverviewBlock and OverviewSection/>', () => {
  const qaHook = 'qaHook';
  it('renders the base overview block with no extra settings passed', () => {
    const testRow = renderer.create(<XUIOverviewBlock />);
    expect(testRow).toMatchSnapshot();
  });
  it('renders extra overview block classes that are passed in', () => {
    const wrapper = mount(<XUIOverviewBlock className="testClass" />);
    const tag = wrapper.find(XUIOverviewBlock);
    expect(tag.hasClass('testClass')).toEqual(true);
  });
  it('renders overview block without default layout', () => {
    const wrapper = mount(<XUIOverviewBlock hasLayout={false} />);
    expect(wrapper.find('.xui-overview-layout').length).toBe(0);
  });
  it('renders the base overview block with sections, children, sentiments, and options', () => {
    const testBlock = renderer.create(
      <XUIOverviewBlock hasBorder={false} hasBackground={false} textAlignment="left">
        {Object.keys(overviewSentiments).map((sentiment, index) => {
          return (
            <XUIOverviewSection key={index} sentiment={sentiment} label={sentiment} value="100.23">
              test child content
            </XUIOverviewSection>
          );
        })}
      </XUIOverviewBlock>,
    );
    expect(testBlock).toMatchSnapshot();
  });
  it('renders the overview sections with proper sentiments', () => {
    Object.keys(overviewSentiments).forEach(sentiment => {
      const section = mount(
        <XUIOverviewSection sentiment={sentiment} label={sentiment} value="100.23" />,
      );
      if (overviewSentiments.sentiment) {
        expect(section.find(`.xui-textcolor-${sentiment}`).length).toBe(1);
      }
    });
  });
  it('renders extra overview section classes that are passed in', () => {
    const wrapper = mount(
      <XUIOverviewBlock>
        <XUIOverviewSection label="labelText" value="65.43" className="testClass" />
      </XUIOverviewBlock>,
    );
    const tag = wrapper.find(XUIOverviewSection);
    expect(tag.hasClass('testClass')).toEqual(true);
  });
  it('renders overview block and overview section with automation id when qaHook prop is passed in', () => {
    const wrapper = renderer.create(
      <XUIOverviewBlock qaHook={qaHook}>
        <XUIOverviewSection qaHook={qaHook} />
      </XUIOverviewBlock>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('XUIOverviewBlock and XUIOverviewSection should pass accessibility testing', async () => {
    const wrapper = mount(
      <XUIOverviewBlock>
        <XUIOverviewSection />
      </XUIOverviewBlock>,
    );
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });
});
