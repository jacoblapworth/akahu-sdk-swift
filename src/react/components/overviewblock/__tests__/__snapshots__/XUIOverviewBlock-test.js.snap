// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<XUI OverviewBlock and OverviewSection/> renders overview block and overview section with automation id when qaHook prop is passed in 1`] = `
<div
  className="xui-overview xui-overview-has-border xui-overview-solid xui-overview-text-align-center xui-overview-layout"
  data-automationid="qaHook"
>
  <section
    className="xui-overview--section"
    data-automationid="qaHook"
  >
    <div
      className="xui-overview--label"
    />
    <div
      className="xui-overview--value"
    />
  </section>
</div>
`;

exports[`<XUI OverviewBlock and OverviewSection/> renders the base overview block with no extra settings passed 1`] = `
<div
  className="xui-overview xui-overview-has-border xui-overview-solid xui-overview-text-align-center xui-overview-layout"
/>
`;

exports[`<XUI OverviewBlock and OverviewSection/> renders the base overview block with sections, children, sentiments, and options 1`] = `
<div
  className="xui-overview xui-overview-text-align-left xui-overview-layout"
>
  <section
    className="xui-overview--section"
  >
    <div
      className="xui-overview--label"
    >
      standard
    </div>
    <div
      className="xui-overview--value"
    >
      100.23
    </div>
    test child content
  </section>
  <section
    className="xui-overview--section"
  >
    <div
      className="xui-overview--label"
    >
      positive
    </div>
    <div
      className="xui-overview--value xui-textcolor-positive"
    >
      100.23
    </div>
    test child content
  </section>
  <section
    className="xui-overview--section"
  >
    <div
      className="xui-overview--label"
    >
      negative
    </div>
    <div
      className="xui-overview--value xui-textcolor-negative"
    >
      100.23
    </div>
    test child content
  </section>
  <section
    className="xui-overview--section"
  >
    <div
      className="xui-overview--label"
    >
      muted
    </div>
    <div
      className="xui-overview--value xui-textcolor-muted"
    >
      100.23
    </div>
    test child content
  </section>
</div>
`;
